tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1501,17 +1501,22 @@ def decode_image(contents, channels=None, name=None):
     contents: 0-D `string`. The encoded image bytes.
     channels: An optional `int`. Defaults to `0`. Number of color channels for
       the decoded image.
+    dtype: The desired `tf.DType` of the returned `Tensor`. Must be either
+      `tf.uint8` or `tf.uint16`.
     name: A name for the operation (optional)
 
   Returns:
-    `Tensor` with type `uint8` with shape `[height, width, num_channels]` for
+    `Tensor` with type `dtype` and shape `[height, width, num_channels]` for
       BMP, JPEG, and PNG images and shape `[num_frames, height, width, 3]` for
       GIF images.
 
   Raises:
     ValueError: On incorrect number of channels.
   """
   with ops.name_scope(name, 'decode_image'):
+    if dtype != dtypes.uint8 and dtype != dtypes.uint16:
```

Comment:
Since `convert_image_dtype` handles more than`uint8` and `uint16`, this can be removed.